src/site/content/en/reliable/two-way-communication-guide/index.md

```diff
@@ -0,0 +1,400 @@
+to send a port to the registered service worker. The page also implements an `onmessage` listener on
+it to receive messages from the other context:
+
+```javascript
+const messageChannel = new MessageChannel();
+
+//Init port
+navigator.serviceWorker.controller.postMessage({type: 'PORT_INITIALIZATION'}, [
+  messageChannel.port2,
+]);
+
+//Listen to messages
+messageChannel.port1.onmessage = (event) => {
+  // Process message
+};
+```
+
+<figure class="w-figure">
+  <img src="message-channel.png"
+       width="600"
+       alt="Diagram showing a page passing a port to a service worker, to establish two-way communication.">
+</figure>
+
+The service worker receives the port, saves a reference to it and uses it to postMessage the other
+side:
+
+```javascript
+let communicationPort;
+
+//Save reference to port
+self.addEventListener('message', (event) => {
+  if (event.data && event.data.type === 'PORT_INITIALIZATION') {
+    communicationPort = event.ports[0];
+  }
+});
+
+//Send messages
+communicationPort.postMessage({type: 'MSG_ID'});
+```
+
+`MessageChannel` is currently supported by [all major
+browsers](https://caniuse.com/?search=channel).
+
+### Advanced APIs: Background Sync and Background Fetch
+
+In this guide we explored ways of implementing **two-way** communication techniques, for relatively
+simple cases, like passing a string message describing the operation to perform, or a list of URLs
+to cache from one context to the other. In this section we'll explore two APIs to handle specific
+scenarios: lack of connectivity and long downloads.
+
+#### Background Sync
+
+A chat app might want to make sure that messages are never lost due to bad connectivity. The
+[Background Sync API](https://developers.google.com/web/updates/2015/12/background-sync) lets you
+defer actions to be retried when the user has stable connectivity. This is useful for ensuring that
+whatever the user wants to send, is actually sent. 
+
+Instead of the `postMessage()` interface, the page register for a `sync`:
+
+```javascript
+navigator.serviceWorker.ready.then(function (swRegistration) {
+  return swRegistration.sync.register('myFirstSync');
+});
+```
+
+The service worker then listens for the `sync` event to process the message:
+
+```javascript
+self.addEventListener('sync', function (event) {
+  if (event.tag == 'myFirstSync') {
+    event.waitUntil(doSomeStuff());
+  }
+});
+```
+
+The function `doSomeStuff()` should return a promise indicating the success/failure of whatever it's
+trying to do. If it fulfills, the sync is complete. If it fails, another sync will be scheduled to
+retry. Retry syncs also wait for connectivity, and employ an exponential back-off. 
+
+Once the operation has been performed, the service worker can then communicate back with the page to
+update the UI, by using any of the communication APIs explored earlier.
+
+Google search uses `Background Sync` to persist failed queries due to bad connectivity, and retry
```

Comment:
Formatting of Background Sync is not consistent with earlier

Reply:
I see in all cases it appears as `Background Sync`. I can't find the mismatch here.

Reply:
Oh, sorry. It's the fact that you formatted it as inline code:

    `Background Sync`

Whereas everywhere else you did not use inline code:

    Background Sync

Reply:
```suggestion
Google search uses Background Sync to persist failed queries due to bad connectivity, and retry
```